pkg/apiserver/webhook/configmap_webhook.go

```diff
@@ -106,24 +110,28 @@ func (v *ConfigMapValidator) validateUpdate(req admission.Request) admission.Res
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	oldCloudResourcePrefix, err := v.getCloudResourcePrefix(oldConfigMap)
+	oldControllerConfig, err := v.getControllerConfig(oldConfigMap)
 	if err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	if newCloudResourcePrefix == oldCloudResourcePrefix {
-		return admission.Allowed("")
-	}
-	if v.NpControllerInterface.IsCloudResourceCreated() {
-		return admission.Denied(fmt.Sprintf("cloud resource is already created with CloudResourcePrefix %s. "+
-			"Please delete the resource and try or use the same CloudResourcePrefix",
-			oldCloudResourcePrefix))
+	if newControllerConfig.CloudResourcePrefix != oldControllerConfig.CloudResourcePrefix {
+		if v.NpControllerInterface.IsCloudResourceCreated() {
+			return admission.Denied(fmt.Sprintf("cloud resource is already created with CloudResourcePrefix %s. "+
+				"Please delete the resource and try or use the same CloudResourcePrefix", oldControllerConfig.CloudResourcePrefix))
+		}
+		if len(newControllerConfig.CloudResourcePrefix) > 0 {
```

Comment:
logically it doesnt seem right.. You are comparing at line 118 and then you are checking with len. Please rework the flow

Reply:
Solved